policy-controller/core/src/http_route.rs

```diff
@@ -176,3 +197,51 @@ impl PartialEq for QueryParamMatch {
 }
 
 impl Eq for QueryParamMatch {}
+
+impl Status {
+    pub fn collect_from(status: Option<gateway::HttpRouteStatus>) -> Vec<Self> {
+        status
+            .into_iter()
+            .map(|status| {
+                status
+                    .inner
+                    .parents
+                    .iter()
+                    .filter_map(|parent_status| Self::from_parent_status(parent_status))
+                    .collect::<Vec<_>>()
+            })
+            .flatten()
+            .collect()
+    }
+
+    fn from_parent_status(status: &gateway::RouteParentStatus) -> Option<Self> {
+        if let Some(ref kind) = status.parent_ref.kind {
+            if kind != "Server" {
+                return None;
+            }
+        }
+
+        let conditions = status
+            .conditions
+            .iter()
+            .filter_map(|condition| {
+                let type_ = match condition.type_.as_ref() {
+                    "Accepted" => ConditionType::Accepted,
+                    "ResolvedRefs" => ConditionType::ResolvedRefs,
+                    _ => return None,
```

Comment:
do we expect to see other condition types that we just don't want to do anything about, or is this an error case? if it's an error, maybe we should log something?

Reply:
Yea this is an error case. The status controller shouldn't be setting types other than `Accepted` or `ResolvedRefs`, so it's probably a good idea to handle this more gracefully and log an error when there is an unexpected status type.